### PR TITLE
Update rules_k8s image and use workload identity

### PR DESCRIPTION
**Pull request by fejta**

### CODE REVIEW

File: config/prow/config.yaml [hunk 1]
```diff
@@ -34,6 +34,34 @@ plank:
         sidecar:
           requests:
             cpu: 100m
+    'bazelbuild/rules_k8s':
```
Comment (chaodaiG) [inline config/prow/config.yaml:37]: There is a nicer way of doing this now:
```suggestion
    default_decoration_config_entries:
    - timeout: 2h
      grace_period: 15m
      utility_images:
        clonerefs: "gcr.io/k8s-prow/clonerefs:v20220609-19e07c7554"
        initupload: "gcr.io/k8s-prow/initupload:v20220609-19e07c7554"
        entrypoint: "gcr.io/k8s-prow/entrypoint:v20220609-19e07c7554"
        sidecar: "gcr.io/k8s-prow/sidecar:v20220609-19e07c7554"
      gcs_configuration:
        bucket: "kubernetes-jenkins"
        path_strategy: "legacy"
        default_org: "kubernetes"
        default_repo: "kubernetes"
      gcs_credentials_secret: "service-account"
      resources:
        clonerefs:
          requests:
            cpu: 100m
        initupload:
          requests:
            cpu: 100m
        place_entrypoint:
          requests:
            cpu: 100m
        sidecar:
          requests:
            cpu: 100m
    - orgRepo: "bazelbuild/rules_k8s"
       config:
         gcs_credentials_secret: ""
         default_service_account_name: "prowjob-default-sa"
```

see https://github.com/GoogleCloudPlatform/oss-test-infra/blob/9ab89d1c145ca2cd06e94faab3f505dda6a6daf9/prow/oss/config.yaml#L104 as an example

**Comment by chaodaiG** [comment]
/hold
in case you want to use my suggestion in this PR

File: config/prow/config.yaml (hunk 1 above)
Comment (fejta) [reply config/prow/config.yaml]: How does this look now?

**Review by cjwagner — APPROVED** [review]

**Pull request merged**